RoboJackets/urc-software
# Feat/rover simulation

# Description
This PR does the following:
- Adds the option of using a simple diff drive robot instead of the complex 6-wheeled rover
- Refactors code related to navigation, setting up urc_navigation as the home of all config and launch files related to localization and navigation
- Ensures functionality of ros2 control and gazebo

Fixes #{{ https://app.clickup.com/t/86az1q5bh }}

# Testing Steps
- colcon build --symlink-install
- . install/setup.bash
- ros2 launch urc_bringup bringup.launch.py

Expectation: This should spin up Gazebo, ROS2 Conrol, RViz, and the EKF node

NOTES:
These fixes need to be made:
[ERROR] [1705871675.218470482] [rviz2]: The link is static or has unrealistic inertia, so the equivalent inertia box will not be shown.
This is because of the inertia on the rover footprint

Gazebo ROS2 Not Running:
The controller manager is not working


# Design Presentation: N/A0

# Self Checklist
- [x ] I have formatted my code using `ament_uncrustify --reformat`
- [ x] I have tested that the new behavior works
